### DISCUSSION TITLE
[Feature Request] Allow customizing the output format of `smartlog`

### DISCUSSION THREAD

**Discussion by mfulbright** [arxanas/git-branchless#1352] [Ideas]
For each revision that `smartlog` displays, it prints some information. I.e. at the moment it prints:
- The short hash of the commit
- The time since the commit was last modified
- Any branches pointing at the commit
- The commit message

It would be cool if `smartlog` had an argument where you could specify a format string that would control what's printed for each revision. E.g. I don't think I really care about the time since the commit was modified. I would love, on the other hand, if I could display a link to a PR on github that's tied to a branch pointing at the revision (though maybe it's a stretch to expect this to be possible).

**Comment by arxanas** [comment]
It certainly makes sense as a feature for `smartlog`, but at some point, you might just prefer to use [jj](https://github.com/martinvonz/jj), which already implements this with its [templating engine](https://martinvonz.github.io/jj/latest/templates/) 🙂.
